From 094bdc96113024ffb2e850a52cb6897001d8dccb Mon Sep 17 00:00:00 2001
From: Stanislaw Kardach <stanislaw.kardach@cavium.com>
Date: Fri, 19 Oct 2018 10:05:52 +0300
Subject: [PATCH 0331/1051] net: octeontx-fpa: fix sysfs VF indexing

pdev->devfn is offset from the pci_iov_virtfn_devfn(fpa->pdev, i) so
the indexing of VF device structures in _show() functions was offset
by 1 which caused invalid values to be shown.

Additionally there was a possibility of a NULL offset read because of
the indexing above.

This commit fixes the indexing offset to retrieve a proper device
structure. This applies for FPA device files: pool_maxcnt, pool_curcnt
and pool_redcnt.

Signed-off-by: Stanislaw Kardach <stanislaw.kardach@cavium.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/fpapf_main.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/fpapf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/fpapf_main.c
index d568cd9bf79a..7dbdcef3ecd3 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/fpapf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/fpapf_main.c
@@ -250,7 +250,7 @@ static ssize_t pool_maxcnt_show(struct device *dev,
 {
 	struct fpapf *curr, *fpa = NULL;
 	struct pci_dev *pdev = container_of(dev, struct pci_dev, dev);
-	int i, n, vfid = pdev->devfn;
+	int i, n, vfid = pdev->devfn - 1;
 	u64 cnt;
 	char info[512];
 
@@ -282,7 +282,7 @@ static ssize_t pool_curcnt_show(struct device *dev,
 {
 	struct fpapf *curr, *fpa = NULL;
 	struct pci_dev *pdev = container_of(dev, struct pci_dev, dev);
-	int i, n, vfid = pdev->devfn;
+	int i, n, vfid = pdev->devfn - 1;
 	u64 cnt;
 	char info[512];
 
@@ -314,7 +314,7 @@ static ssize_t pool_redcnt_show(struct device *dev,
 {
 	struct fpapf *curr, *fpa = NULL;
 	struct pci_dev *pdev = container_of(dev, struct pci_dev, dev);
-	int i, n, vfid = pdev->devfn;
+	int i, n, vfid = pdev->devfn - 1;
 	u64 reg, ena, lvl, pass, drop, aura;
 	char *info;
 
-- 
2.17.1

